src/sysext/sysext.c

```diff
@@ -533,15 +533,18 @@ static int merge_subprocess(Hashmap *images, const char *workspace) {
                                                "Extension image contains /usr/lib/os-release file, which is not allowed (it may carry /etc/os-release), refusing.");
 
                 /* Now that we can look into the extension image, let's see of the OS version is compatible */
-                r = parse_os_release(
+                r = parse_extension_release(
                                 p,
+                                img->name,
                                 "ID", &extension_os_release_id,
                                 "VERSION_ID", &extension_os_release_version_id,
                                 "SYSEXT_LEVEL", &extension_os_release_sysext_level,
                                 NULL);
-                if (r == -ENOENT)
-                        log_notice_errno(r, "Extension '%s' carries no os-release data, not checking for version compatibility.", img->name);
-                else if (r < 0)
+                if (r == -ENOENT){
+                        log_notice("Extension '%s' carries no extension-release data, ignoring extension.", img->name);
```

Comment:
please use log_notice_errno() here, as before.

if you have an error code, pass it to the logging functions, no matter what. Even if you don't include `%m` in the log string. Thing is that the error code is also passed along as structured log info to the journal, and it's useful to pass any error we got there, so the the data there is complete

Reply:
oh, and `if (…){` → `if (…) {`

Reply:
ah yes certainly, got mangled in the rebase+change